src/main/java/org/reactome/release/common/dataretrieval/cosmic/COSMICFileRetriever.java

```diff
@@ -0,0 +1,122 @@
+package org.reactome.release.common.dataretrieval.cosmic;
+
+import org.reactome.release.common.dataretrieval.AuthenticatableFileRetriever;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Base64;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+/**
+ * Get COSMIC data file.
+ * 
+ * The COSMIC data file cannot be downloaded in a single step.
+ * To download the COSMIC data file programmatically, you must first request the file 
+ * with a Basic Authorization header which includes the username and password, base64 encoded.
+ * You will get a small JSON document back, if authorization succeeds. This document will contain
+ * a "url" element. The data file can be accessed if you download the URL that is returned in the JSON
+ * document.
+ * @author sshorser
+ *
+ */
+public class COSMICFileRetriever extends AuthenticatableFileRetriever
+{
+	public COSMICFileRetriever(String retrieverName)
+	{
+		super(retrieverName);
+	}
+	
+	public COSMICFileRetriever()
+	{
+		super();
+	}
+	
+	private boolean retrieveAndSetCOSMICDownloadURL() throws UnsupportedEncodingException
+	{
+		boolean gotDownloadURLOK = false;
+		// The NEW process to download from COSMIC requires a few steps.
+		// 1) Generate a base64-encoded string of the username and password.
+		// 2) Send this encoded string in the "Authorization: Basic" header to the URL of the file you want.
+		// 3) Parse the response - extract the "url" JSON attribute and then download THAT URL to get the REAL file.
+		
+		// Encoded string.
+		String encodedUsernamePassword = Base64.getEncoder().encodeToString((this.userName + ":" + this.password).getBytes("UTF-8"));
+		
+		HttpGet get = new HttpGet(this.uri);
+		//Need to multiply by 1000 because timeouts are in milliseconds.
+		int delayInMilliseconds = 1000 * (int)this.timeout.getSeconds();
```

Comment:
I'm a bit confused - the `getSeconds()` either returns seconds, and so would need to be _divided_ by 1000 to get it in milliseconds, or it returns milliseconds and so would need to be _multiplied_ by a 1000 to get second, and so the variable would need to be called `delayInSeconds`?

Reply:
Oh wow nevermind I think I need a coffee